src/controllers/controllerinputmappingtablemodel.cpp

```diff
@@ -47,6 +47,12 @@ void ControllerInputMappingTableModel::onPresetLoaded() {
             m_midiInputMappings = m_pMidiPreset->getInputMappings().values();
             endInsertRows();
         }
+        connect(this,
+                &QAbstractTableModel::dataChanged,
+                this,
+                [this]() {
+                    m_pMidiPreset->setDirty(true);
+                });
```

Comment:
note that `dataChanged()` is emitted also when an non/existing vlaues are 'confirmed' with Return. IMO this is the only minor regression _if users are not used to this default QAbstarctItemView/QSpinBox/QTestEdit behaviour_, though it's now consistent with the existing GUI (skins, preferences).